### PR TITLE
Omit needless words in track tutorial

### PR DESCRIPTION
**Pull request by danielmitterdorfer**

### CODE REVIEW

File: docs/adding_tracks.rst [hunk 1]
```diff
@@ -530,26 +527,26 @@ Finally, specify the compressed file name in your ``track.json`` file in the ``s
 
 Specifying ``compressed-bytes`` (file size of ``documents.json.bz2``) and ``uncompressed-bytes`` (file size of ``documents.json``) is optional but helps Rally to provide progress indicators and also verify integrity.
 
-You've now mastered the basics of track development for Rally. It's time to pat yourself on the back before you dive into the advanced topics!
+You've now mastered the basics of track development for Rally. It's time to pat yourself on the back before you dive into the advanced topics.
```
Comment (dliappis) [inline docs/adding_tracks.rst]: I must say I liked the use of an exclamation mark here! I know https://developers.google.com/style/exclamation-points discourages it, but we are not calling attention to an important point, but rather, lighten the mood and "reward" the reader, after digging into a deep topic.

File: docs/adding_tracks.rst [hunk 2]
```diff
@@ -530,26 +527,26 @@ Finally, specify the compressed file name in your ``track.json`` file in the ``s
...
 
 Advanced topics
 ---------------
 
 Template Language
 ^^^^^^^^^^^^^^^^^
 
-Rally uses `Jinja2 <http://jinja.pocoo.org/docs/dev/>`_ as template language. This allows you to use Jinja2 expressions in track files.
+Rally uses `Jinja2 <http://jinja.pocoo.org/docs/dev/>`_ as template language so you can use Jinja2 expressions in track files.
```
Comment (dliappis) [inline docs/adding_tracks.rst]: as **a** template language?

File: docs/adding_tracks.rst [hunk 3]
```diff
@@ -530,26 +527,26 @@ Finally, specify the compressed file name in your ``track.json`` file in the ``s
...
 
 
 Extension Points
 """"""""""""""""
 
 Rally also provides a few extension points to Jinja2:
```
Comment (dliappis) [inline docs/adding_tracks.rst]: How about just `a few extensions` (without points)?

File: docs/adding_tracks.rst [hunk 4]
```diff
@@ -568,7 +565,7 @@ You can find an example in the http_logs track::
       }
     }
 
-The data set that is used in the http_logs track starts on 26-04-1998 but we want to ignore the first few days for this query, so we start on 15-05-1998. The expression ``{{'15-05-1998' | days_ago(now)}}`` yields the difference in days between now and the fixed start date and allows us to benchmark time range queries relative to now with a predetermined data set.
+The data set that is used in the ``http_logs`` track starts on 26-04-1998 but we want to ignore the first few days for this query, so we start on 15-05-1998. The expression ``{{'15-05-1998' | days_ago(now)}}`` yields the difference in days between now and the fixed start date and allows us to benchmark time range queries relative to now with a predetermined data set.
```
Comment (dliappis) [inline docs/adding_tracks.rst]: Substitute `yields` with just provides (or returns)? Not every track author may be familiar with Python's `yield` and generators.

File: docs/adding_tracks.rst [hunk 5]
```diff
@@ -605,7 +602,7 @@ First, define the name of your parameter source in the operation definition::
       "professions": ["mechanic", "physician", "nurse"]
     }
 
-Rally will recognize the parameter source and looks then for a file ``track.py`` in the same directory as the corresponding JSON file. This file contains the implementation of the parameter source::
+Rally recognizes the parameter source and looks for a file ``track.py`` next to ``track.json``. This file contains the implementation of the parameter source::
```
Comment (dliappis) [inline docs/adding_tracks.rst]: Replace `next` with:

> in the same directory (or folder) as ``track.json``.

File: docs/adding_tracks.rst [hunk 6]
```diff
@@ -786,11 +775,11 @@ Then create a file ``track.py`` next to ``track.json`` and implement the followi
 The function ``percolate`` is the actual runner and takes the following parameters:
 
 * ``es``, which is the Elasticsearch Python client
```
Comment (dliappis) [inline docs/adding_tracks.rst]: I think `which` can be skipped

File: docs/adding_tracks.rst [hunk 7]
```diff
@@ -786,11 +775,11 @@ Then create a file ``track.py`` next to ``track.json`` and implement the followi
...
-* ``params`` which is a dict of parameters provided by its corresponding parameter source. Treat this parameter as read only and do not attempt to write to it.
+* ``params`` which is a ``dict`` of parameters provided by its corresponding parameter source. Treat this parameter as read-only.
```
Comment (dliappis) [inline docs/adding_tracks.rst]: Again `which` I think is not needed.

File: docs/adding_tracks.rst [hunk 8]
```diff
@@ -212,19 +208,17 @@ Congratulations, you have created your first track! You can test it with ``esral
 Adding support for test mode
 ----------------------------
 
-When you invoke Rally with ``--test-mode``, it switches to a mode that allows you to check your track very quickly for syntax errors. To achieve that, it will postprocess its internal track representation after loading it:
-
-* Iteration-based tasks will run at most one warmup iteration and one measurement iteration.
-* Time-period-based tasks will run for at most 10 seconds without any warmup.
+You can check your track very quickly for syntax errors when you invoke Rally with ``--test-mode``. Rally postprocesses its internal track representation then:
```
[the remark this answers is not in the document]
Comment (danielmitterdorfer) [reply docs/adding_tracks.rst]: How about this?

> Rally postprocesses its internal track representation as follows:

File: docs/adding_tracks.rst [hunk 9]
```diff
@@ -212,19 +208,17 @@ Congratulations, you have created your first track! You can test it with ``esral
...
 
-Rally will postprocess all data file names of a track. So instead of ``documents.json``, Rally will attempt to find ``documents-1k.json`` and will assume it contains 1.000 documents. However, you need to prepare these data files otherwise this test mode is not supported.
+* Iteration-based tasks run at most one warmup iteration and one measurement iteration.
+* Time-period-based tasks run for at most 10 seconds without warmup.
```
[the remark this answers is not in the document]
Comment (danielmitterdorfer) [reply docs/adding_tracks.rst]: I interpret the "at most" as a restrictive clause in this case. Hence we should not enclose it in commas. So I'd write: "run at most for"

File: docs/adding_tracks.rst [hunk 10]
```diff
@@ -175,11 +175,7 @@ Finally, add a file called ``track.json`` in the tutorial directory::
     }
 
 
-A few things to note:
-
-* The numbers below the ``documents`` property are needed to verify integrity and provide progress reports. Determine the correct document count with ``wc -l documents.json`` and the size in bytes with ``stat -f "%z" documents.json``.
-* You can add as many queries as you want. We use the `official Python Elasticsearch client <http://elasticsearch-py.readthedocs.org/>`_ to issue queries.
-
+The numbers below the ``documents`` property are needed to verify integrity and provide progress reports. Determine the correct document count with ``wc -l documents.json`` and the size in bytes with ``stat -f "%z" documents.json``.
```
[the remark this answers is not in the document]
Comment (danielmitterdorfer) [reply docs/adding_tracks.rst]: Changed.

File: docs/adding_tracks.rst [hunk 11]
```diff
@@ -409,7 +403,7 @@ If you want to reuse operation definitions across challenges, you can also defin
 
 Note how we reference to the operations by their name (e.g. ``create``, ``bulk-index``, ``force-merge`` or ``query-match-all``).
 
-If your track consists of multiple challenges, it can be cumbersome to include them all explicitly. Therefore Rally brings a ``collect`` helper that collects all related files for you. Let's adapt our track to use it::
+It can be cumbersome to include multiple challenges explicitly so you can use Rally's ``collect`` helper instead::
```
[the remark this answers is not in the document]
Comment (danielmitterdorfer) [reply docs/adding_tracks.rst]: Changed.

File: docs/adding_tracks.rst [hunk 12]
```diff
@@ -474,39 +468,42 @@ If your track consists of multiple challenges, it can be cumbersome to include t
...
 .. note::
 
     Rally's log file contains the fully rendered track after it has loaded it successfully.
 
 .. note::
 
-    If you define multiple challenges, Rally will run the challenge where ``default`` is set to ``true``. If you want to run a different challenge, provide the command line option ``--challenge=YOUR_CHALLENGE_NAME``.
+    If you define multiple challenges, Rally runs the challenge where ``default`` is set to ``true``. If you want to run a different challenge, provide the command line option ``--challenge=YOUR_CHALLENGE_NAME``.
 
 You can even use `Jinja2 variables <http://jinja.pocoo.org/docs/2.9/templates/#assignments>`_ but you need to import the Rally helpers a bit differently then. You also need to declare all variables before the ``import`` statement::
```
[the remark this answers is not in the document]
Comment (danielmitterdorfer) [reply docs/adding_tracks.rst]: Makes sense. I've changed it.

File: docs/adding_tracks.rst (hunk 1 above)
Comment (danielmitterdorfer) [reply docs/adding_tracks.rst]: I reverted my change.

File: docs/adding_tracks.rst (hunk 3 above)
Comment (danielmitterdorfer) [reply docs/adding_tracks.rst]: That's simpler indeed. I also changed the heading accordingly.

File: docs/adding_tracks.rst (hunk 4 above)
Comment (danielmitterdorfer) [reply docs/adding_tracks.rst]: I use the word "yield" here in a mathematical, not in a software engineering context.

File: docs/adding_tracks.rst (hunk 5 above)
Comment (danielmitterdorfer) [reply docs/adding_tracks.rst]: I think "next to X" is just as clear?

File: docs/adding_tracks.rst (hunk 6 above)
Comment (danielmitterdorfer) [reply docs/adding_tracks.rst]: Makes sense. I've changed it.

File: docs/adding_tracks.rst (hunk 7 above)
Comment (danielmitterdorfer) [reply docs/adding_tracks.rst]: Makes sense. I've changed it.

File: docs/adding_tracks.rst (hunk 8 above)
[the remark this answers is not in the document]
Comment (dliappis) [reply docs/adding_tracks.rst]: Sounds good.

**Pull request merged**